src/StyledWindowPortal.tsx

```diff
@@ -93,12 +94,26 @@ class StyledWindowPortal extends React.PureComponent<Props, State> {
         }
       }
     );
+
+    if (this.props.autoClose) {
+      window.addEventListener("unload", this.closeExternalWindow);
+    }
   }
 
   componentWillUnmount() {
-    if (!!this.state.externalWindow) this.state.externalWindow.close();
+    if (this.props.autoClose) {
+      window.removeEventListener("unload", this.closeExternalWindow);
+    }
```

Comment:
Is there a case here where if the component is mounted with autoClose = true and is subsequently changed to false before unmounting then this removeEventListener is not called?

Removing the conditional here would mean the event listener is removed if it exists and does not harm if it does not.